### PR TITLE
Enable Win32-OpenSSH autostart and fix sshd_config for OpenSSH < 8.2

### PR DESCRIPTION
**Pull request by fw2568**
The `SshModule` (OS-level Win32-OpenSSH daemon, RFC 0018) had three defects — the reported one plus two the new e2e suite uncovered.

**1. A found sshd was never set to autostart.** `InstallOpenSshServerAsync` sets the service Automatic, but no-ops when sshd is already installed — so a *found* daemon left at Manual/Disabled was configured but never came back after a reboot. New idempotent `EnsureSshdAutostartAsync` (Automatic + start) runs whenever the module owns an installed daemon.

**2. `Include` in sshd_config breaks the inbox sshd.** The module wrote `Include sshd_config.d/*.conf`, but `Include` in `sshd_config` requires OpenSSH 8.2+; the Windows-inbox sshd is 8.1p1 on Server 2022 and rejects it (`Bad configuration option: Include`), so sshd failed to start — ssh was effectively broken on those images. The directives are now inlined into the main `sshd_config` in a marked managed block (works on all sshd versions), replacing the `Include` + drop-in mechanism. A legacy `Include` line is stripped on re-provision.

**3. Host-key generation hangs against a running sshd.** A running daemon holds the `ssh_host_*_key` files open, so `RegenerateSshHostKeys`' pre-delete fails and `ssh-keygen` blocks on an "overwrite?" prompt. The module now stops sshd (`StopSshdAsync`) before host-key generation, and enables autostart last (after keys exist).

Also: fingerprint reporting fires before the autostart step (so a rotated key is surfaced even if autostart fails), and an unterminated managed-block marker no longer discards the rest of `sshd_config`.

Adds the `SshServer` e2e suite (asserts sshd ends Running + Automatic, and is re-enabled when found at Manual/Stopped and when found already Running) plus unit coverage (`BuildSshdManagedConfig` edge cases, stop→regenerate→autostart ordering, DryRun interception). RFC 0018 updated. Verified: 841 unit tests + 5/5 e2e green on a winsrv2022 catlet.

### CODE REVIEW

File: src/Eryph.GuestServices.Provisioning/Modules/SshModule.cs [hunk 1]
```diff
@@ -66,17 +64,24 @@ public async Task<ModuleOutcome> ApplyAsync(
             return ModuleOutcome.Ok();
         }
 
+        // 2. Stop sshd before touching host keys. A daemon that is ALREADY
+        // running (a found image with sshd enabled at boot) holds the
+        // ssh_host_*_key files open; RegenerateSshHostKeys' pre-delete then
+        // fails and ssh-keygen hangs on an "overwrite?" prompt. Stopping is
+        // idempotent; the restart/autostart below brings it back up.
+        await context.Os.StopSshdAsync(cancellationToken).ConfigureAwait(false);
+
```
[the remark this answers is not in the document]
Comment (fw2568) [reply src/Eryph.GuestServices.Provisioning/Modules/SshModule.cs]: Fixed in aa850d1 — the stop step is now inside the try/catch, so a stop failure returns `ModuleOutcome.Fail` (and skips host-key generation rather than risking the lock hang). Added a `Stop_failure_returns_Fail_and_skips_host_key_generation` test.

File: src/Eryph.GuestServices.Provisioning/Modules/SshModule.cs [hunk 2]
```diff
@@ -66,17 +64,24 @@ public async Task<ModuleOutcome> ApplyAsync(
...
         try
         {
-            // 2. Host keys.
+            // 3. Host keys.
             (hostKeysChanged, reportableFingerprints) =
                 await ProcessHostKeysAsync(config, context, cancellationToken).ConfigureAwait(false);
 
-            // 3. Drop-in sshd_config.
+            // 4. sshd_config directives.
             configChanged = await ProcessDropInAsync(config, context, cancellationToken).ConfigureAwait(false);
```
[the remark this answers is not in the document]
Comment (fw2568) [reply src/Eryph.GuestServices.Provisioning/Modules/SshModule.cs]: Renamed `ProcessDropInAsync` -> `ProcessSshdConfigAsync` in aa850d1 to match the inlined managed-block behavior.

File: src/Eryph.GuestServices.Provisioning/Modules/SshModule.cs [hunk 3]
```diff
@@ -213,17 +237,16 @@ await context.Os.WriteSshHostKeyAsync(
...
 
     /// <summary>
-    /// Builds and writes the <c>50-eryph.conf</c> drop-in. Returns true if a
-    /// drop-in with at least one directive was written.
+    /// Builds the eryph directive block and writes it into the main
+    /// <c>sshd_config</c> (see <see cref="IWindowsOs.WriteSshdManagedConfigAsync"/>
+    /// for why it is inlined rather than an <c>Include</c> drop-in). Returns true
+    /// if the config file changed (so the caller restarts sshd).
     /// </summary>
     private async Task<bool> ProcessDropInAsync(
```
[the remark this answers is not in the document]
Comment (fw2568) [reply src/Eryph.GuestServices.Provisioning/Modules/SshModule.cs]: Done — renamed to `ProcessSshdConfigAsync` (aa850d1).

File: src/Eryph.GuestServices.Provisioning/Windows/WindowsOs.cs [hunk 4]
```diff
@@ -1113,27 +1133,79 @@ private async Task<SshHostKeyFingerprint> ReadHostKeyFingerprintAsync(
...
+    // rejects it and refuses to start — so we no longer use it and strip any
+    // copy we previously wrote when re-provisioning such a guest.
+    private const string LegacySshdIncludeDirective = "Include sshd_config.d/*.conf";
+
+    // internal for unit testing — pure form of the sshd_config write. Prepends
+    // our directives in a marked, managed block so their first-obtained-value
+    // settings win over shipped directives (and the trailing
+    // `Match Group administrators` block). We inline the directives directly
+    // instead of an `Include` drop-in because `Include` is unsupported by
+    // OpenSSH &lt; 8.2 (the inbox Windows sshd), which then fails to start.
```
[the remark this answers is not in the document]
Comment (fw2568) [reply src/Eryph.GuestServices.Provisioning/Windows/WindowsOs.cs]: Fixed in 82172f5 — this is a plain `//` comment, so it now uses the literal `<`.

File: src/Eryph.GuestServices.Provisioning/Windows/DryRunWindowsOs.cs [hunk 5]
```diff
@@ -177,18 +183,17 @@ public Task WriteSshHostKeyAsync(
...
     }
 
-    public Task EnsureSshdConfigIncludeAsync(CancellationToken cancellationToken)
+    public Task<bool> WriteSshdManagedConfigAsync(string directives, CancellationToken cancellationToken)
     {
-        logger.LogInformation("DRY-RUN would ensure sshd_config Include directive for sshd_config.d");
-        return Task.CompletedTask;
+        logger.LogInformation(
+            "DRY-RUN would write eryph-managed sshd_config directives ({Bytes} bytes)",
+            directives.Length);
```
[the remark this answers is not in the document]
Comment (fw2568) [reply src/Eryph.GuestServices.Provisioning/Windows/DryRunWindowsOs.cs:194]: Fixed in 82172f5 — the dry-run log now says `chars` (it's `directives.Length`, a character count).
